Sources/Overload/OvEditor/src/OvEditor/Panels/Hierarchy.cpp

```diff
@@ -282,10 +282,8 @@ void OvEditor::Panels::Hierarchy::AddActorByInstance(OvCore::ECS::Actor & p_acto
 	textSelectable.AddPlugin<OvUI::Plugins::DDSource<std::pair<OvCore::ECS::Actor*, OvUI::Widgets::Layout::TreeNode*>>>("Actor", "Attach to...", std::make_pair(&p_actor, &textSelectable));
 	textSelectable.AddPlugin<OvUI::Plugins::DDTarget<std::pair<OvCore::ECS::Actor*, OvUI::Widgets::Layout::TreeNode*>>>("Actor").DataReceivedEvent += [&p_actor, &textSelectable](std::pair<OvCore::ECS::Actor*, OvUI::Widgets::Layout::TreeNode*> p_element)
 	{
-		if (p_element.second->HasParent())
-			p_element.second->GetParent()->UnconsiderWidget(*p_element.second);
-
-		textSelectable.ConsiderWidget(*p_element.second);
+		if (p_actor.IsDescendantOf(p_element.first))
+			return;
```

Comment:
nit: instead of returning if the drop destination is a descendant of the drag source, we could perform the parenting if the drop destination is not a descendant of the drag source:

```
if (!p_actor.IsDescendantOf(p_element.first))
{
    p_element.first->SetParent(p_actor);
}
```

Could be nice to warn the user about why the operation is being aborted, so maybe:

```
if (!p_actor.IsDescendantOf(p_element.first))
{
    p_element.first->SetParent(p_actor);
}
else
{
    OVLOG_WARNING("Cannot attach \"" + p_element.first->GetName() + "\" to \"" + p_actor.GetName() + "\" because it is a descendant of the latter.");
}
```

Reply:
Warning added in the latest commit.